libs/iterator_support/include/hpx/iterator_support/traits/is_sentinel_for.hpp

```diff
@@ -0,0 +1,61 @@
+//  Copyright (c) 2020 Giannis Gonidelis
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <hpx/type_support/always_void.hpp>
+#include <hpx/iterator_support/traits/is_sentinel_for.hpp>
+
+#include <utility>
+
+
+namespace hpx { namespace traits {
+
+    template <typename Sent, typename Iter, typename Enable = void>
+    struct is_sentinel_for : std::false_type
+    {
+    };
+
+    template <typename Sent, typename Iter>
+    struct is_sentinel_for<Sent, Iter,
+        typename util::always_void<
+            typename is_iterator<Iter>::type,
+            typename detail::equality_result<Iter, Sent>::type,
+            typename detail::equality_result<Sent, Iter>::type,
+            typename detail::inequality_result<Iter, Sent>::type,
+            typename detail::inequality_result<Sent, Iter>::type>::type>
+      : std::true_type
+    {
+    };
+
+#if defined(HPX_HAVE_CXX20_DISABLE_SIZED_SENTINEL_FOR)
+    template <typename Sent, typename Iter>
+    constexpr bool disable_sized_sentinel_for =
+        std::disable_sized_sentinel_for<Sent, Iter>;
+#else
+    template <typename Sent, typename Iter>
+    constexpr bool disable_sized_sentinel_for = false;
+#endif
+
+    template <typename Sent, typename Iter, typename Enable = void>
+    struct sized_sentinel_for : std::false_type
+    {
+    };
+
+    template <typename Sent, typename Iter>
+    struct sized_sentinel_for<Sent, Iter,
+        typename util::always_void<
+            typename hpx::traits::is_sentinel_for<Sent, Iter>::type,
```

Comment:
`hpx::traits::is_sentinel_for<Sent, Iter>::type` is always value (as it evaluates to either `std::true_type` or `std::false_type`). You might want to include this into the `enable_if` condition below.